Install powertop into dom0 and enable autotune service at boot.

tlaurion opened the pull request:
Install with `sudo qubesctl state.sls my-config.powertop`

tlaurion committed:
Install powertop into dom0 and enable autotune service at boot.

SkypLabs commented on my-config.top:
```diff
@@ -9,6 +9,7 @@ base:
     - my-config.personal-pub
     - my-config.untrusted
     - my-config.vault
+    - my-config.dom0-powertop 
```
Can you rename this file to `powertop.sls`? It's better to use generic names since this state can be applied to other targets (see [this issue](https://github.com/SkypLabs/vim-config/issues/1) for example).

SkypLabs commented on my-config/dom0-powertop.sls:
```diff
@@ -0,0 +1,10 @@
+# -*- coding: utf-8 -*-
+# vim: set syntax=yaml ts=2 sw=2 sts=2 et :
+
+dom0:
```
It's cleaner to write it this way:

```
powertop:
  pkg.installed:
    - fromrepo: fedora
  service.running:
    - enable: True
```

tlaurion replied:
In fact, we want the server to be enabled, but not running. Powertop is not a service, but oneshoot. It is ran and then die. When launched how you suggest, the formula fails because the service is dead when checked. 

It works as it is.

SkypLabs commented on my-config/dom0-powertop.sls:
```diff
@@ -0,0 +1,10 @@
+    - name: powertop
+    - fromrepo: fedora
+powertop:
+  service.dead:
```
`service.dead` will stop the process even with the `enable: True` option. I suppose you meant to use `service.running` instead.

tlaurion replied:
Powertop is not a service, but oneshoot. It is ran and then die. When launched how you suggest, the formula fails because the service is dead when checked.

SkypLabs replied:
> Powertop is not a service, but oneshoot.

Indeed, I forgot this. Consequently, why not using `service.enabled`?

I suggest you to rewrite the state this way:

```
powertop:
  pkg.installed:
    - fromrepo: fedora
  service.enabled: []
```

Then, I will accept your changes.

SkypLabs reviewed (changes requested):
Thanks for this PR.

Can you do the changes suggested in my review before merging your work?

tlaurion committed:
Rename dom0-powertop.sls to powertop.sls

tlaurion committed:
Update my-config.top

Change dom0-powertop to powertop

tlaurion commented:
Because if set like that, the formula fails. This is why I check for dead,
which means enabled but not running ( any other state then running)

Le mar. 10 avr. 2018 13:05, Skyper <notifications@github.com> a écrit :

> *@SkypLabs* commented on this pull request.
> ------------------------------
>
> In my-config/dom0-powertop.sls
> <https://github.com/SkypLabs/my-qubes-os-formula/pull/2#discussion_r180497461>
> :
>
> > @@ -0,0 +1,10 @@
> +# -*- coding: utf-8 -*-
> +# vim: set syntax=yaml ts=2 sw=2 sts=2 et :
> +
> +dom0:
> +  pkg.installed:
> +    - name: powertop
> +    - fromrepo: fedora
> +powertop:
> +  service.dead:
>
> Powertop is not a service, but oneshoot.
>
> Indeed, I forgot this. Consequently, why not using service.enabled?
>
> I suggest you to rewrite the state this way:
>
> powertop:
>   pkg.installed:
>     - fromrepo: fedora
>   service.enabled: []
>
> Then, I will accept your changes.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/SkypLabs/my-qubes-os-formula/pull/2#discussion_r180497461>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAygsk3M5u0EX71VrczFn9U8G3Y9SATIks5tnOZVgaJpZM4TKK_A>
> .
>

SkypLabs commented:
> Because if set like that, the formula fails. This is why I check for dead, which means enabled but not running ( any other state then running)

It works fine for me when using `service.enabled` (from dom0):

```
[skyper@dom0 ~]$ sudo systemctl status powertop
● powertop.service - PowerTOP autotuner
   Loaded: loaded (/usr/lib/systemd/system/powertop.service; enabled; vendor preset: disabled)
   Active: inactive (dead)
```

As you can see, the service is well enabled after executing the state as it is written in my comment. `service.enabled` doesn't care if the service is running or dead. It just enables it.

tlaurion committed:
Update powertop.sls

Changed to state to be enabled only. Didn't knew enabled: [] was doing the right thing!

tlaurion reviewed (commented):
You were right :) Thanks!

tlaurion reviewed (commented):
Done

tlaurion committed:
Update powertop.sls

Corrections

SkypLabs committed:
fix state's id

SkypLabs committed:
fix alphabetical order of the states

SkypLabs commented:
Thanks again :+1:
